pkg/vm/engine/tae/moengine/block.go

```diff
@@ -45,9 +45,9 @@ func (blk *txnBlock) Read(attrs []string, compressed []*bytes.Buffer, deCompress
 		}
 		view.ApplyDeletes()
 		if view.GetData().Allocated() > 0 {
-			bat.Vecs[i] = containers.CopyToMoVector(view.GetData())
+			bat.Vecs[i] = containers.CopyToMoVec(view.GetData())
```

Comment:
What is the different between CopyToMoVec and UnmarshalToMoVec?   Which one does not copy?   Which one share the underlying data?

How will bat.Vecs[i] know who manages the data?   Ever need to free?

Reply:
UbnarshalToMoVec shares the underlying memory and CopyToXXX does copy.

In this scenario, the application layer will pass a buffer in. When the data of the returned tae vector is on this buffer, and no additional memory is allocated, it can share memory. Once new additional memory is allocated, it will be copied to avoid memory life cycle management. 

There are probably these use scenarios:
1. The application layer provide a buffer and TaeVec data store the underlying data on the buffer. Both the TaeVec and MoVec share the underlying memory on the buffer. In this case, TaeVec and MoVec are only two different Readers and Writers, neither owns the memory. In the above use case, when TaeVec.Allocated() == 0, neither TaeVec nor the converted MoVec owns the memory, and neither needs to free the memory
2. When TaeVec owns the memory, and we just want to execute some different read-only operation on the data, we can create a new `reader` like MoVec. For example, when we flush a column data, we can do this:
    ```
    moVec := taeVec.UnmarshalToVec(xxx)
    buf := moVec.Show()
    writer.Write(buf)
    ```

Reply:
I don't know how these work -- can both threads read and write at the same time?
Unless both cases they are read only.